cli/src/pcluster/templates/budget_builder.py

```diff
@@ -0,0 +1,71 @@
+from aws_cdk import aws_budgets as budgets
+from aws_cdk.core import Construct
+
+from pcluster.config.cluster_config import BaseClusterConfig
+
+
+class CostBudgets:
+    """Create the budgets based on the configuration file."""
+
+    def __init__(self, scope: Construct, cluster_config: BaseClusterConfig):
+        self.cluster_config = cluster_config
+        self.scope = scope
+
+    def create_budgets(self):
```

Comment:
The `create_budgets` function is long and contains many things, let's break it into several small functions.
First, let's rename it to _add_resources(), as we do for other builder stack files.
Then you can think about breaking the function into:
Add budget_data, add notifications_with_subscribers...
Or think about breaking it into: add_cluster_budget, add_queue_budget, add_custom_budget...

Reply:
Good call! Restructured it.